lib/wallaroo/core/initialization/local_topology.pony

```diff
@@ -1254,7 +1274,7 @@ actor LocalTopologyInitializer is LayoutInitializer
     match _topology
     | let t: LocalTopology =>
       let current_worker_count = t.worker_names.size()
-      let new_t = t.add_new_worker(joining_worker_name)
+      let new_t = local_topology_for_joining_worker(joining_worker_name)
```

Comment:
You have the topology here, so you can pass it in to `local_topology_for_joining_worker` directly.  That will avoid the need to do the match and handle the else in that method.

Reply:
so would the following suffice:

```
let new_t = local_topology_for_joining_worker(joining_worker_name)
      let current_worker_count = new_t.worker_names.size()
      _router_registry.worker_join(conn, joining_worker_name,
        joining_worker_count, new_t, current_worker_count)
```

Reply:
The idea is to change:
```
  fun ref local_topology_for_joining_worker(
    joining_worker_name: String): LocalTopology
  =>
    match _topology
    | let t: LocalTopology =>
      t.add_new_worker(joining_worker_name)
        .new_barrier_source_id(_routing_id_gen())
    else
      Unreachable()
      LocalTopology("", "", recover val Dag[StepInitializer] end,
        recover val Map[U128, SetIs[RoutingId] val] end,
        recover val Array[WorkerName] end, recover val SetIs[WorkerName] end,
        recover val Map[RoutingId, Map[WorkerName, RoutingId] val] end,
        0)
    end
```
to
```
  fun ref local_topology_for_joining_worker(t: LocalTopology,
    joining_worker_name: String): LocalTopology
  =>
    t.add_new_worker(joining_worker_name)
      .new_barrier_source_id(_routing_id_gen())
```
And then the only change to the line you reference is:
```
let new_t = local_topology_for_joining_worker(t, joining_worker_name)
```
since we've already matched in that context and know `t` is a `LocalTopology`.

Reply:
This way if there's more that needs to be updated for a joining worker, we handle that in one function.